cloud/rackspace/rax_facts.py

```diff
@@ -98,6 +98,11 @@ def rax_facts(module, address, name, server_id):
         except Exception, e:
             pass
 
+    #rackspace likes to leave servers in the deleted state for a while.  
+    #If you make a server with the same name it triggers len(servers) > 1 
+    #so just remove servers with a server status of deleted from the list
+    servers = [server for server in servers if server.status != "DELETED"]
```

Comment:
I'm not a huge fan of re-assigning already assigned variables.  My preference would be a normal for loop with a del.

Otherwise a new variable.